core/src/banking_stage/consumer.rs

```diff
@@ -1065,6 +1092,161 @@ mod tests {
         Blockstore::destroy(ledger_path.path()).unwrap();
     }
 
+    #[test]
+    fn test_bank_process_and_record_transactions_in_last_tick() {
```

Comment:
can we name the test to more clearly state that it is enforcing the nonce update blockhash query ordering rather than a more generic process that could have other side effects that we're _not_ testing here?  something like...
```suggestion
    fn test_bank_nonce_update_blockhash_queried_before_transaction_record() {
```